objstore.go

```diff
@@ -85,6 +85,9 @@ type BucketReader interface {
 	// IsObjNotFoundErr returns true if error means that object is not found. Relevant to Get operations.
 	IsObjNotFoundErr(err error) bool
 
+	// IsCustomerManagedKeyError returns true if the permissions for key used to encrypt the object was revoked.
+	IsCustomerManagedKeyError(err error) bool
```

Comment:
Could we return a newly defined `ErrInvalidCustomerManagedKey` error and use `errors.Cause` to identify it? This way a client does not need an instance of the bucket in order to distinguish between error types.

Reply:
I missed that we already have `IsObjNotFoundErr`, so I guess this should also be fine.